### PR TITLE
Fix Upsert config validation to check for metrics aggregation

### PR DESCRIPTION
**Pull request by geeknarrator**
Bugfix for: https://github.com/apache/pinot/issues/8226

### CODE REVIEW

**Review by Jackie-Jiang — COMMENTED** [review]
I don't think it can compile now because of the extra back brace. Can you please also fix the code format using [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#setup-ide)?

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java [hunk 1]
```diff
@@ -534,6 +534,19 @@ static void validateUpsertConfig(TableConfig tableConfig, Schema schema) {
...
+  }
+
+  /**
+      * Checks if Metrics aggregation is enabled.
+      */
+  @VisibleForTesting
+     static boolean isAggregateMetricsEnabled(TableConfig config) {
+      if(config.getIndexingConfig() == null || config.getIngestionConfig() == null) {return false;
+      }
+      return config.getIndexingConfig().isAggregateMetrics() && CollectionUtils.isEmpty(config.getIngestionConfig().getAggregationConfigs());
```
Comment (Jackie-Jiang) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java:549]: This should be `||` instead of `&&`

Comment (geeknarrator) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java]: I think I referred `MutableSegmentImpl` and misunderstood. Fixed it.

**Comment by geeknarrator** [comment]
> I don't think it can compile now because of the extra back brace. Can you please also fix the code format using [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#setup-ide)?

Thanks, I configured Pinot style.

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java [hunk 2]
```diff
@@ -521,19 +520,32 @@ static void validateUpsertConfig(TableConfig tableConfig, Schema schema) {
...
+
+  /**
+   * Checks if Metrics aggregation is enabled.
+   */
+  @VisibleForTesting
+  static boolean isAggregateMetricsEnabled(TableConfig config) {
+    boolean isAggregateMetricsEnabledInIndexingConfig =
+        Optional.ofNullable(config.getIndexingConfig()).map(IndexingConfig::isAggregateMetrics).orElse(false);
+    boolean hasAggregationConfigs = Optional.ofNullable(config.getIngestionConfig())
+        .map(ingestionConfig -> CollectionUtils.isNotEmpty(ingestionConfig.getAggregationConfigs())).orElse(false);
```
[the remark this answers is not in the document]
Comment (geeknarrator) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java:547]: 👍 sgtm

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java [hunk 3]
```diff
@@ -521,19 +520,32 @@ static void validateUpsertConfig(TableConfig tableConfig, Schema schema) {
...
+    return isAggregateMetricsEnabledInIndexingConfig || hasAggregationConfigs;
```
[the remark this answers is not in the document]
Comment (geeknarrator) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java:548]: I see.. I didn't know that is a requirement.

[the remark this answers is not in the document]
Comment (geeknarrator) [reply pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java]: Done.

**Comment by geeknarrator** [comment]
@Jackie-Jiang - Thanks for reviewing the changes :) I have addressed the remaining comments. Once you approve those, can I get it merged? I don't seem to have the access.

**Comment by geeknarrator** [comment]
@Jackie-Jiang - Could you please help me merge this? I have resolved the conflict. (also the formatting changes are done by the pinot checkstyle configuration.

**Pull request merged**